db/src/db.rs

```diff
@@ -689,130 +762,67 @@ impl DB {
-                    };
-
-                    let a: i64 = match a_ {
-                        &entmod::Entid::Entid(ref a__) => *a__,
-                        &entmod::Entid::Ident(ref a__) => *self.schema.require_entid(&a__.to_string())?,
-                    };
-
-                    let attribute: &Attribute = self.schema.require_attribute_for_entid(&a)?;
-                    if attribute.fulltext {
-                        bail!(ErrorKind::NotYetImplemented(format!("Transacting :db/fulltext entities is not yet implemented: {:?}", entity)))
-                    }
-
-                    // This is our chance to do schema-aware typechecking: to either assert that the
-                    // given value is in the attribute's value set, or (in limited cases) to coerce
-                    // the value into the attribute's value set.
-                    let typed_value: TypedValue = self.to_typed_value(v_, &attribute)?;
-
-                    let added = true;
-                    if attribute.multival {
-                        non_fts_many.push((e, a, typed_value, added));
-                    } else {
-                        non_fts_one.push((e, a, typed_value, added));
-                    }
-                    Ok(())
-                },
-
-                Entity::AddOrRetract {
-                    op: OpType::Retract,
-                    e: entmod::EntidOrLookupRef::Entid(ref e_),
-                    a: ref a_,
-                    v: entmod::ValueOrLookupRef::Value(ref v_) } => {
-
-                    let e: i64 = match e_ {
-                        &entmod::Entid::Entid(ref e__) => *e__,
-                        &entmod::Entid::Ident(ref e__) => *self.schema.require_entid(&e__.to_string())?,
-                    };
-
-                    let a: i64 = match a_ {
-                        &entmod::Entid::Entid(ref a__) => *a__,
-                        &entmod::Entid::Ident(ref a__) => *self.schema.require_entid(&a__.to_string())?,
-                    };
-
-                    let attribute: &Attribute = self.schema.require_attribute_for_entid(&a)?;
-                    if attribute.fulltext {
-                        bail!(ErrorKind::NotYetImplemented(format!("Transacting :db/fulltext entities is not yet implemented: {:?}", entity)))
-                    }
-
-                    // This is our chance to do schema-aware typechecking: to either assert that the
-                    // given value is in the attribute's value set, or (in limited cases) to coerce
-                    // the value into the attribute's value set.
-                    let typed_value: TypedValue = self.to_typed_value(v_, &attribute)?;
-
-                    let added = false;
+    /// Update the current partition map materialized view.
+    // TODO: only update changed partitions.
+    pub fn update_partition_map(&self, conn: &rusqlite::Connection) -> Result<()> {
```

Comment:
I haven't really worked out the details, but I think there's an entirely different division of responsibilities in the future.  Right now, the data structures `DB` and `Tx` "own" the work of updating the SQLite store and processing an incoming transaction.  My vision is that there's a split between the "updating the SQLite gubbins" and "term rewriting, resolution, etc" on top.

Following your suggestion, we might indeed define type
```rust
type SQLiteConnection = &mut rusqlite::Connection;
```
and a trait that defines the "low level interface", like
```rust
trait ConcreteRepresentation {
  pub fn lookup_avs(...);
  pub fn insert_non_fts_one(...);
  ...
}
```
Then, we might
```rust
impl ConcreteRepresentation for SQLiteConnection {
  pub fn lookup_avs(...) { // Using SQLite-specific features, etc.
  ...
}
```

That would separate the `DB` type from the implementation details that are specific to SQLite.  You could imagine that some `ConcreteRepresentation` implementations might not support fulltext indexing at all; or that there's an implementation that uses Postgres or some other backing store.

Does that seem sensible?